### PR TITLE
[Survey][Instrument_List] Allow instruments from the battery to be set as surveys

### PR DESCRIPTION
**Pull request by ZainVirani**
This PR addresses one small part of the design change documented [here](https://forum.loris.ca/t/feature-proposal-enable-adding-surveys-from-instrument-list-in-candidate-timepoint/157).

Specifically it allows instruments in a battery for a timepoint that has been started to be set as a survey in the survey module (and optionally sent to a candidate/proband in an email).
In the instrument_list module, the link to the instrument will not appear for surveys *until* they have been set as surveys in the survey module (there is a tooltip explaining this to users who hover over the name of the survey in the instrument list).

### CODE REVIEW

**Comment by ZainVirani** [comment]
This won't work right now, I need to make a slight change.

**Comment by kongtiaowang** [comment]
Fixing the PHAN errors.

**Comment by HenriRabalais** [comment]
@ZainVirani How should this be tested?

File: modules/instrument_list/php/instrument_list.class.inc [hunk 1]
```diff
@@ -185,6 +207,10 @@ class Instrument_List extends \NDB_Menu_Filter
 
                 $Ins = "instruments";
 
+                if ($isSurvey == '1' && !$isSent) {
```
Comment (HenriRabalais) [inline modules/instrument_list/php/instrument_list.class.inc]: Should the value for `$isSurvey` be expected as a string?

File: modules/instrument_list/php/instrument_list.class.inc [hunk 2]
```diff
@@ -167,6 +169,26 @@ class Instrument_List extends \NDB_Menu_Filter
...
+                    WHERE Test_name=:TN
+                    AND CommentID=:CID",
+                    array(
+                     'TN'  => $instrument['Test_name'],
+                     'CID' => $instrument['CommentID'],
+                    )
+                );
+
                 // print the sub group header row
                 if ($instrument['Sub_group'] != $prevSubgroup) {
```
Comment (HenriRabalais) [inline modules/instrument_list/php/instrument_list.class.inc]: given that `$instrument['Sub_group'] ` is set to `$prevSubgroup` I think you can add an extra equal sign (`!==`) for a type check as well.

Comment (HenriRabalais) [reply modules/instrument_list/php/instrument_list.class.inc]: Sorry, just realised this change was not made by you. Ignore that!

File: modules/instrument_list/php/instrument_list.class.inc (hunk 1 above)
Comment (ZainVirani) [reply modules/instrument_list/php/instrument_list.class.inc]: You're right, it's a `tinyint(1)`

Comment (ZainVirani) [reply modules/instrument_list/php/instrument_list.class.inc]: It's `tinyint(1)` in the DB but `gettype($isSurvey)` returns string. I think this check is fine.

**Comment by HenriRabalais** [comment]
Attempted a manual test, but my battery is not loaded due to the fact that I do not have instruments. I recommend that this be tested by someone else.

**Comment by ZainVirani** [comment]
@christinerogers @me-evans I'm going to close this. It doesn't seem like any decision has been made and I haven't heard anything about this since May.

**Pull request closed without merging**